docs/en/operation/Server-Logging.md

```diff
@@ -91,6 +91,107 @@ $ ./bin/alluxio logLevel --logName=alluxio.heartbeat.HeartbeatContext --target=w
+in `logs/fuse.log`:
+
+```
+2020-03-03 14:33:35,129 DEBUG AlluxioFuseFileSystem - Enter: chmod(path=/aaa,mode=100644)
+2020-03-03 14:33:35,131 DEBUG AlluxioFuseFileSystem - Exit (0): chmod(path=/aaa,mode=100644) in 2 ms
+2020-03-03 14:33:35,132 DEBUG AlluxioFuseFileSystem - Enter: getattr(path=/aaa)
+2020-03-03 14:33:35,135 DEBUG AlluxioFuseFileSystem - Exit (0): getattr(path=/aaa) in 3 ms
+2020-03-03 14:33:35,138 DEBUG AlluxioFuseFileSystem - Enter: getattr(path=/._aaa)
+2020-03-03 14:33:35,140 DEBUG AlluxioFuseFileSystem - Failed to get info of /._aaa, path does not exist or is invalid
+2020-03-03 14:33:35,140 DEBUG AlluxioFuseFileSystem - Exit (-2): getattr(path=/._aaa) in 2 ms
+```
+
+### Logging RPCs Calls Sent by Client
+
+Add to your application-side `log4j.properties` to capture RPCs between Alluxio client
+and FileSystem Master:
+
+```properties
+log4j.logger.alluxio.client.file.FileSystemMasterClient=DEBUG
+```
+
+Similarly, capture lower-level RPCs between Alluxio client and Block Master:
+
+```properties
+log4j.logger.alluxio.client.block.BlockMasterClient=DEBUG
+```
+
+You will see debug logs on the begin and the end of each RPC with its arguments and result
+in client logs like the following:
+
+```
+2020-03-03 15:56:40,115 DEBUG FileSystemMasterClient - Enter: GetStatus(path=/.DS_Store,options=loadMetadataType: ONCE
+commonOptions {
+  syncIntervalMs: -1
+  ttl: -1
+  ttlAction: DELETE
+}
+)
+2020-03-03 15:56:40,117 DEBUG FileSystemMasterClient - Exit (ERROR): GetStatus(path=/.DS_Store,options=loadMetadataType: ONCE
+commonOptions {
+  syncIntervalMs: -1
+  ttl: -1
+  ttlAction: DELETE
+}
+) in 2 ms: alluxio.exception.status.NotFoundException: Path "/.DS_Store" does not exist.
+```
+
+### Logging RPC Calls Received by Masters
+
+On master, one can turn on the debug-level RPC logging for File System level RPC calls (e.g.,
+creating/reading/writing/removing files, updating file attributions) using `logLevel` command:
+
+```console
+$ ./bin/alluxio logLevel \
+--logName=alluxio.master.file.FileSystemMasterClientServiceHandler \
+--target master --level=DEBUG
+```
+
+Similarly, turn on the debug-level logging for Block related RPC calls (e.g., adding/removing
+blocks):
+
+```console
+# For Block Master logging:
+$ ./bin/alluxio logLevel \
+--logName=alluxio.master.block.BlockMasterClientServiceHandler \
+--target master --level=DEBUG
+```
+
+### Identifying Expensive Client RPCs / FUSE calls
+
+When debugging the performance, it is often useful to understand which RPCs take most of the time
+but without recording all the communication (e.g., enabling all debug logging). There are two
+properties introduced in v2.3.0 in Alluxio to record expensive calls or RPCs in logs with WARN
+level. For example, setting the following in `conf/alluxio-site.properties` will record client-side
+RPCs taking more than 200ms and FUSE APIs taking more than 1s:
```

Comment:
Please put each sentence on a new line in this paragraph

Reply:
fixed